src/app/container/container.component.ts

```diff
@@ -173,8 +179,14 @@ export class ContainerComponent extends Entry implements AfterViewInit {
         this.published = this.tool.is_published;
         if (this.tool.workflowVersions.length === 0) {
           this.selectedVersion = null;
+          this.versionsFileTypes = [];
         } else {
           this.selectedVersion = this.selectTag(this.tool.workflowVersions, this.urlVersion, this.tool.defaultVersion);
+          if (this.selectedVersion) {
+            this.entryService.getVersionsFileTypes(tool.id, this.selectedVersion.id).subscribe((fileTypes: Array<SourceFile.TypeEnum>) => {
```

Comment:
At a brief glance, it seems like we're not very good/consistent on this, but ideally you would handle the error case too. This comment applies to several places.

Reply:
Handled error cases by adding alerts and setting variable to empty array here and in other locations.